cpp/lib/src/FileUtil.cc

```diff
@@ -1476,4 +1491,81 @@ bool GetMostRecentlyModifiedFile(const std::string &directory_path, std::string
 }
 
 
+void RemoveLeadingBytes(const std::string &path, const loff_t no_of_bytes) {
+    if (no_of_bytes < 0)
+        LOG_ERROR("no_of_bytes must not be negative!");
+
+    if (no_of_bytes == 0)
+        return;
+
+    const auto original_file_size(GetFileSize(path));
+    if (original_file_size <= no_of_bytes) {
+        if (::truncate(path.c_str(), 0) != 0)
+            LOG_ERROR("failed to truncate(2) \"" + path + "\" to 0!");
+        return;
+    }
+
+    const int fd(::open(path.c_str(), O_LARGEFILE|O_RDWR));
+    if (fd == -1)
+        LOG_ERROR("failed to open(2) \"" + path + "\"!");
+
+    void * const map_start(::mmap(nullptr, original_file_size, PROT_READ|PROT_WRITE, MAP_SHARED, fd, 0));
+    if (map_start == MAP_FAILED or map_start == nullptr)
+        LOG_ERROR("Failed to mmap(2) \"" + path + "\"!");
+
+    ::memmove(map_start, static_cast<void *>(static_cast<char *>(map_start) + no_of_bytes), original_file_size - no_of_bytes);
```

Comment:
What happens if the log file is written between mmap and memmove?

Reply:
We're up shit creek or, I have to go back to the drawing board! :-(